New warning in cyvcf2 causes tests to fail

```
_____________________ test_vcf_to_zarr__empty_region[True] _____________________

shared_datadir = PosixPath('/tmp/pytest-of-runner/pytest-0/test_vcf_to_zarr__empty_region0/data')
is_path = True
tmp_path = PosixPath('/tmp/pytest-of-runner/pytest-0/test_vcf_to_zarr__empty_region0')

    @pytest.mark.parametrize(
        "is_path",
        [True, False],
    )
    def test_vcf_to_zarr__empty_region(shared_datadir, is_path, tmp_path):
        path = path_for_test(shared_datadir, "CEUTrio.20.21.gatk3.4.g.vcf.bgz", is_path)
        output = tmp_path.joinpath("vcf_concat.zarr").as_posix()
        regions = "23"
    
>       vcf_to_zarr(path, output, regions=regions)

sgkit/tests/io/vcf/test_vcf_reader.py:319: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
sgkit/io/vcf/vcf_reader.py:923: in vcf_to_zarr
    field_defs=field_defs,
sgkit/io/vcf/vcf_reader.py:425: in vcf_to_zarr_sequential
    for variants_chunk in chunks(region_filter(variants, region), chunk_length):
sgkit/io/vcf/utils.py:33: in chunks
    for first in iterator:  # take one item out (exits loop if `iterator` is empty)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   UserWarning: no intervals found for b'/tmp/pytest-of-runner/pytest-0/test_vcf_to_zarr__empty_region0/data/CEUTrio.20.21.gatk3.4.g.vcf.bgz' at 23

cyvcf2/cyvcf2.pyx:456: UserWarning
```

Caused by https://github.com/brentp/cyvcf2/pull/229. This fix is very simple, just add a pytest filter warnings.

--- a/sgkit/tests/io/vcf/test_vcf_reader.py
+++ b/sgkit/tests/io/vcf/test_vcf_reader.py
@@ -311,6 +311,7 @@ def test_vcf_to_zarr__parallel(shared_datadir, is_path, output_is_path, tmp_path
     "is_path",
     [True, False],
 )
+@pytest.mark.filterwarnings("ignore::UserWarning")
 def test_vcf_to_zarr__empty_region(shared_datadir, is_path, tmp_path):
     path = path_for_test(shared_datadir, "CEUTrio.20.21.gatk3.4.g.vcf.bgz", is_path)
     output = tmp_path.joinpath("vcf_concat.zarr").as_posix()

